### PR TITLE
refactor(plot): refactor hover in plot package

### PR DESCRIPTION
**Pull request by atsushi421**
## Description
This PR refactors hover.py in plot package.

## Related links
To reproduce:
- [ipynb](https://drive.google.com/drive/folders/1k7SliaXZeguHYcqMXPr2wmK-C0qrucOw?usp=share_link)
- [archtecture file](https://drive.google.com/file/d/1W9PDyj046xzapMA-p3o6HWtUhMA8JhIG/view?usp=share_link)
- [trace data](https://drive.google.com/file/d/1xy1L4c-eGi9RSHog1qNgovTcwxoM74Z9/view?usp=share_link)

## Notes for reviewers
I have confirmed that there are no regressions using CARET_report.
image

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] (Optional) The PR has been properly tested with CARET_report verification.
- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/caret_analyze/plot/visualize_lib/bokeh/util/hover.py [hunk 1]
```diff
@@ -138,18 +153,123 @@ def create(self, options: dict = {}) -> HoverTool:
...
+
+    def _validate(self, target_object: Any) -> None:
+        if not isinstance(target_object, (CallbackBase, Communication, Publisher, Subscription)):
+            raise InvalidArgumentError(
+                "'target_object' must be [CallbackBase/Communication/Publisher/Subscription]"
+                'in timeseries graph.'
+            )
+
+    def to_list(self) -> List[str]:
+        if isinstance(self._target_object, CallbackBase):
```
Comment (keita1523) [inline src/caret_analyze/plot/visualize_lib/bokeh/util/hover.py:200]:
```suggestion
    def to_list(self) -> List[str]:
        hover_keys: List[str] = []
        if isinstance(self._target_object, CallbackBase):
```

**Review by keita1523 — CHANGES_REQUESTED** [review]
The refactored structure is good.
Please modify easy review.

File: src/caret_analyze/plot/visualize_lib/bokeh/util/hover.py (hunk 1 above)
Comment (atsushi421) [reply src/caret_analyze/plot/visualize_lib/bokeh/util/hover.py]: Fixed in https://github.com/tier4/CARET_analyze/pull/277/commits/3082601e1601048319240a444cf61997f1a65bf3

**Review by keita1523 — APPROVED** [review]
LGTM

**Comment by atsushi421** [comment]
@takeshi-iwanari 
Could you please review this PR?

**Review by takeshi-iwanari — APPROVED** [review]
LGTM!

**Pull request merged**